FaunaDB.Client/Types/Codec.cs

```diff
@@ -47,6 +47,33 @@ public static IResult<Value> VALUE(Value value)
 
         public static IResult<IReadOnlyDictionary<string, Value>> OBJECT(Value input) =>
             Cast.MapTo<ObjectV, IReadOnlyDictionary<string, Value>>(input, x => x.Value);
+
+        public static IResult<T> DECODE<T>(Value input)
+        {
+            try
+            {
+                return Success(Decoder.Decode<T>(input));
+            }
+            catch (InvalidOperationException ex)
+            {
+                return Fail<T>(ex.Message);
+            }
+        }
+
+        public static Func<Value, IResult<object>> DECODE(Type dstType)
```

Comment:
I was reviewing this PR and I was thinking to remove that codec but was too late, you started to submit comments 😄 

This was for cases when you don't have the type at compile time, only on runtime. But yes, it might not be correctly pollute the codec api with this. If the user would like to do this they could use the `Decoder` directly.